File: fixtures/projects/e2e-app/src/api/v1/handler.ts
import type { ApiResponse } from "../../types/api";
import { HTTP_STATUS } from "../../constants/http";

export type Status = "idle" | "processing" | "completed" | "failed";

export type ApiV1Context = {
	requestId: string;
	method: string;
	path: string;
};

export function handleRequest(ctx: ApiV1Context): ApiResponse<string> {
	if (ctx.method !== "GET") {
		return { success: false, error: { code: HTTP_STATUS.METHOD_NOT_ALLOWED, message: "Only GET supported in v1" } };
	}
	return { success: true, data: `v1:${ctx.requestId}:${ctx.path}` };
}

export function formatV1Response(data: string): string {
	return `[APIv1] ${data}`;
}

File: fixtures/projects/e2e-csharp/Assets/Scripts/API/V1/Handler.cs
using MyApp.Types;

namespace MyApp.API.V1
{
    public class Handler
    {
        public ApiResponse HandleRequest(string route)
        {
            if (route == "/status")
            {
                return new ApiResponse(true, "v1-online", 200);
            }

            return new ApiResponse(false, "v1-missing", 404);
        }

        public string VersionTag()
        {
            return "v1";
        }
    }
}
